app/Validators/ProfileValidator.ts

```diff
@@ -0,0 +1,42 @@
+import { schema, CustomMessages, rules } from "@ioc:Adonis/Core/Validator";
+import type { HttpContextContract } from "@ioc:Adonis/Core/HttpContext";
+
+class ProfileValidator {
+  constructor(protected ctx: HttpContextContract) {}
+
+  public schema = schema.create({
+    name: schema.string({}, [rules.minLength(3), rules.maxLength(30)]),
+
+    mobileNumber: schema.string({}, [
+      rules.required(),
+      rules.regex(/^[0-9]{10}$/),
+      rules.minLength(10),
+      rules.maxLength(10),
+    ]),
+
+    gender: schema.enum(["MALE", "FEMALE"], [rules.required()]),
+
+    dateOfBirth: schema.date({
```

Comment:
let store birthdate in timestamp format, but will show use "dd-MM-yyyy" format in API response